src/RankMovie.jsx

```diff
@@ -69,39 +68,105 @@ function RankMovie() {
+    if (!currentUser || !newMovie || mid === undefined) {
+      console.error("Current user or new movie data is not available.");
+      return;
+    }
+
+    const response = await fetch(`http://localhost:3000/users/getUserMovie`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        userId: currentUser.uid,
+        movieId: newMovie.movieId,
+      }),
+    });
+    const data = await response.json();
+    setNewUserMovie(data);
+
+    let newLow = low;
+    let newHigh = high;
+
+    if (prefersNewMovie) {
+      newLow = mid + 1;
+    } else {
+      newHigh = mid - 1;
+    }
+    // Clamp Bounds
+    newLow = Math.max(0, newLow);
+    newHigh = Math.min(userMovies.length - 1, newHigh);
 
+    let newDynamicRating;
+    if (prefersNewMovie) {
+      if (mid === userMovies.length - 1) {
+        // New movie is better than the best movie
+        newDynamicRating = 10
+      } else if(low==high){
+        const nextIndex = mid + 1;
+        newDynamicRating = (userMovies[mid].rating + userMovies[nextIndex].rating) / 2;
+      }else{
+        newDynamicRating = (userMovies[newLow].rating + userMovies[newHigh].rating) / 2;
+        console.log("new rating: "+newDynamicRating+ "lower Bound: " + userMovies[newLow].title+ userMovies[newLow].rating+ "Upper Bound: " + userMovies[newHigh].title+ userMovies[newHigh].rating)
+
+      }
+    } else {
+      if (mid === 0) {
+        // New movie is worse than the worst movie
+        newDynamicRating = 1;
+      } else if(low==high) {
+        const prevIndex = mid - 1;
+        newDynamicRating = (userMovies[mid].rating + userMovies[prevIndex].rating) / 2;
+      }else{
+        newDynamicRating = (userMovies[newLow].rating + userMovies[newHigh].rating) / 2;
+        console.log("new rating: "+newDynamicRating+ "lower Bound: " + userMovies[newLow].title+ userMovies[newLow].rating+ "Upper Bound: " + userMovies[newHigh].title+ userMovies[newHigh].rating)
+
+      }
+
+    }
+    let winner, loser;
+    if (prefersNewMovie) {
+      winner = data;
+      loser = userMovies[mid];
+    } else {
+      winner = userMovies[mid];
+      loser = data;
+    }
     const K = 1;
     const divisor = 100;
-    const winnerExpected =
-      1 / (1 + Math.pow(10, (loser.rating - winner.rating) / divisor));
+    const winnerExpected = 1 / (1 + Math.pow(10, (loser.rating - winner.rating) / divisor));
     const loserExpected = 1 - winnerExpected;
     let winnerNewRating = winner.rating + K * (1 - winnerExpected);
     let loserNewRating = loser.rating - K * loserExpected;
 
     winnerNewRating = Math.max(1, Math.min(10, winnerNewRating));
     loserNewRating = Math.max(1, Math.min(10, loserNewRating));
 
-    try {
+    setLow(newLow);
+    setHigh(newHigh);
+
+    // Update ratings for both winner and loser
+    if (prefersNewMovie){
+    await updateMovieRatings(winner.movieId, newDynamicRating);
```

Comment:
This line is not indented. Not sure if it's the only one, but these type of issues should be resolved with the linter. Maybe we can even hook it up to always run when a commit is being made.